src/Silex/Provider/SecurityServiceProvider.php

```diff
@@ -101,6 +101,18 @@ public function register(Application $app)
             });
         }
 
+        $app['user'] = function ($app) {
```

Comment:
this should not be a shared method. Otherwise you will get the wrong value if you access it first before the firewall gets call and then a second time after (the second time will get the cached ``null`` value)